lib/cli/ui/prompt.rb

```diff
@@ -48,35 +55,47 @@ class << self
         # Free form question when the answer can be empty
         #   CLI::UI::Prompt.ask('What is your opinion on this question?', allow_empty: true)
         #
-        # Question with answers
-        #   CLI::UI::Prompt.ask('What kind of project is this?', options: %w(rails go ruby python))
+        # Interactive (multiple choice) question
+        #   CLI::UI::Prompt.ask_interactive('What kind of project is this?', options: %w(rails go ruby python))
         #
+        # Interactive (multiple choice) question with defined handlers
+        #   CLI::UI::Prompt.ask_interactive('What kind of project is this?', options: %w(rails go ruby python)) do |handler|
+        #     handler.option('rails')  { |selection| puts selection } => outputs "rails" if selected
+        #     handler.option('go')     { |selection| puts selection } => outputs "go" if selected
+        #     handler.option('ruby')   { |selection| puts selection } => outputs "ruby" if selected
+        #     handler.option('python') { |selection| puts selection } => outputs "python" if selected
         #
-        def ask(question, options: nil, default: nil, is_file: nil, allow_empty: true)
-          if (default && !allow_empty) || (options && (default || is_file))
-            raise(ArgumentError, 'conflicting arguments')
-          end
+        def ask(question, options: nil, default: nil, is_file: nil, allow_empty: true, &options_proc)
+          raise(ArgumentError, 'conflicting arguments') if ((options || block_given?) && (default || is_file))
```

Comment:
it would be nice to have a better error message now like `explicit options conflicts with default and is_file`